prov/psm2/src/psmx2.h

```diff
@@ -799,7 +747,7 @@ static inline void psmx2_cntr_inc(struct psmx2_fid_cntr *cntr)
 	cntr->counter++;
 	psmx2_cntr_check_trigger(cntr);
 	if (cntr->wait)
-		psmx2_wait_signal((struct fid_wait *)cntr->wait);
+		cntr->wait->signal(cntr->wait);
 }
```

Comment:
Btw, we need to add counters to the utility code.  I'm just not quite sure of the best way to do so without introducing overhead.